active/001-tabular-msa.md

```diff
@@ -0,0 +1,292 @@
+- `items() -> iterator`
+
+   Iterator of `(key, Sequence)` tuples in same order as `keys()`.
+- `iter_positions(reverse=False) -> iterator`
+
+   Iterator of *exactly* `Sequence` objects in the order of positions in the
+   alignment. If `reverse=True`, yields positions in reverse order.
+- `append(Sequence) -> None`
+
+   Add a sequence to the end of the alignment. Must be the same `dtype` and
+   length must match second dimension.
+- `extend(iterable) -> None`
+
+   Add sequences to the end of the alignment. Each sequence must be the same
+   `dtype` and length must match second dimension.
+- `sort(key=None, reverse=False) -> None`
+
+   Sort sequences in-place. If `key=None` the existing keys will be used to
+   sort. Otherwise `key` matches the constructor.
+- `merge(UNRESOLVED) -> TabularMSA`
+
+   Merge two `TabularMSA` objects.
+- `consensus() -> Sequence`
+
+   Return majority consensus sequence matching alignment's `dtype`. Other
+   types of consensus may be added later.
+- `gap_frequencies(axis, relative=False) -> np.ndarray`
+
+   Return vector of gap frequencies along an axis.
+- `entropies(axis) -> np.ndarray`
+
+   Return vector of Shannon entropies along an axis.
+- `reindex(key=None) -> None`
+
+   Assign new keys to sequences in the alignment. `key` matches constructor.
+
+# Drawbacks
+
+- scikit-bio users will have to update their code to use a new API.
+
+- The new class name `TabularMSA` may not be as obvious as to its behavior as
+the current class name `Alignment`.
+
+- It is unclear if it is advisable to use a tabular (row-column) MSA as it
+forces an ordering where none exists. See
+[Lee et al. 2002](http://bioinformatics.oxfordjournals.org/content/18/3/452.short).
+
+# Alternatives
+
+We considered representing an MSA internally as a DAG in support of
+partial-order MSAs. However, representing positions in such an MSA defeats the
+entire purpose of using a DAG at all.
+
+We considered *only* supporting partial-order MSAs. However, since scikit-bio
+is a bioinformatics data munging library, we need to support row-column MSAs
+because most bioinformatics tools use or produce these structures.
+
+We considered the following alternative names for the new class:
+
+- `LinearMSA`: it is not immediately obvious that this is a row-column
+alignment
+- `TotalOrderMSA`: even less obvious and it sounds "better" than partial-order
+if users aren't clear on the definitions
+- `RowColMSA` / `RowColumnMSA`: painful to type and hard to remember (was it
+"col" or "column"?)
+- `StandardAlignment`: implies "best" or preferred alignment structure
+- `Alignment`: too general as it doesn't describe an MSA, also implies a
+default alignment structure
+
+# Unresolved questions
+
+Is it useful to merge two `TabularMSA` objects? What does this interface look
+like? Can both axes be merged? If so, do we support the standard SQL join
+operations? Should it be called `join` instead of `merge`?
```

Comment:
If we do support this, it probably makes most sense for it to do NW alignment of the two ``TablularMSA`` objects. Since that operation will be possible by calling ``nw_align`` directly, it might make more sense to not have a ``merge`` so users are aware they they're running an O(n^2) operation when merging.